### PR TITLE
Remove many job e2e tests.

### PR DESCRIPTION
**Pull request by spxtr**
These tests have equivalent unit test coverage as far as I can tell, in [pkg/controller/job/jobcontroller_test.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/jobcontroller_test.go). See #40839 for context.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/job.go [hunk 1]
```diff
@@ -96,120 +92,6 @@ var _ = framework.KubeDescribe("Job", func() {
...
-		waitForReplicas := kubectl.NewRetryParams(5*time.Second, 5*time.Minute)
-		err = scaler.Scale(f.Namespace.Name, job.Name, uint(endParallelism), nil, waitForScale, waitForReplicas)
-		Expect(err).NotTo(HaveOccurred())
-
-		By("Ensuring active pods == endParallelism")
-		err = waitForAllPodsRunning(f.ClientSet, f.Namespace.Name, job.Name, endParallelism)
-		Expect(err).NotTo(HaveOccurred())
-	})
-
-	It("should delete a job", func() {
```
Comment (soltysh) [inline test/e2e/job.go]: I'd like this one to be left until we have a server deletion in-place and coverage. IIRC this is not tested anywhere else.

**Review by soltysh — CHANGES_REQUESTED** [review]
I'm ok with 4 of the 5 proposed here. Reaping has no coverage afaik.

File: test/e2e/job.go (hunk 1 above)
Comment (spxtr) [inline test/e2e/job.go]: Done.

**Review by soltysh — APPROVED** [review]

**Pull request merged**